src/lib.rs

```diff
@@ -1,3 +1,69 @@
+//! A library that provides `print!`, `println!`, `dbg!` implementations and
+//! logging capabilities for Espressif devices.
+//!
+//! - Supports all Espressif devices.
+//! - Supports different communication methods:
+//!     - UART (Default)
+//!     - JTAG-Serial (Only available in ESP32-C3, ESP32-C6, ESP32-H2, ESP32-S3)
+//!     - [RTT]
+//!     - No-op: Turns printing into a no-op
+//! - Supports [`defmt`] backend
+//!
+//! # Using `esp-println` as a Library:
+//!
+//! ```toml
+//! esp-println = { version = "0.7.0", features = ["esp32c2"] }
+//! ```
+//! or `cargo add esp-println --features esp32c2`
+//!
+//! Its important to specify your target as feature.
+//!
+//! # Features
+//!
+//! - There is one feature for each supported target: `esp32`, `esp32c2`,
+//!   `esp32c3`, `esp32c6`, esp32h2`, `esp32s2`, `esp32s3` and `esp8266`.
+//!    - One of these features must be enabled.
+//!    - Only one of these features can be enabled at a time.
+//! - There is one feature for each supported communication method: `uart`,
+//!   `jtag-serial` and `rtt` `no-op`.
+//!     - Only one of these features can be enabled at a time.
+//! - `critical-section` enables critical sections.
+//! - `colors` enable colored output.
+//! - There is one feature for each [`defmt` supported encoding]:
+//!    - `defmt`: Uses [rzCOBS] encoding and adds framing. This is intended to
+//!      be used with [`espflash`] (See `-L/--log-format` argument of `flash` or
+//!      `monitor` subcommands of `espflash` and `cargo-espflash`)
+//!    - `defmt-raw`: Raw defmt output without additional framing.
```

Comment:
```suggestion
//! - There are two ways to use this library with [`defmt`]:
//!    - `defmt`: This is intended to be used with [`espflash`]
//!      See `-L/--log-format` argument of `flash` or `monitor`
//!      subcommands of `espflash` and `cargo-espflash`). Uses
//!      [rzCOBS] encoding and adds framing.
//!    - `defmt-raw`: Raw defmt output without additional framing.
//!      ⚠️ Be careful when using this feature: you must only write
//!      output using `defmt` macros, or you may irrecoverably corrupt
//!      the output stream! This means that even the bootloader's output
//!      must be disabled.
```

Reply:
Sorry I was a bit lazy with the previous comment here. This change would make the ``[`defmt` supported encoding]`` link unused.